src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1626,15 +1626,15 @@ private void EmitInstanceCallExpression(BoundCall call, UseKind useKind)
                 }
                 else
                 {
-                    // calling a method defined in a base class.
+                    // calling a method defined in a base class or interface.
 
                     // When calling a method that is virtual in metadata on a struct receiver, 
                     // we use a constrained virtual call. If possible, it will skip boxing.
                     if (method.IsMetadataVirtual())
                     {
                         // NB: all methods that a struct could inherit from bases are non-mutating
                         //     treat receiver as ReadOnly
-                        tempOpt = EmitReceiverRef(receiver, AddressKind.ReadOnly);
+                        tempOpt = EmitReceiverRef(receiver, methodContainingType.IsInterface ? AddressKind.Writeable : AddressKind.ReadOnly);
```

Comment:
>methodContainingType.IsInterface [](http://example.com/codeflow?start=60&length=32)

Why only in this case? See a comment above

Reply:
Thanks much for suggesting the scenario where an override is added to the struct. 
That showed that the assumption was broken. Removed it.